vpr/src/route/rr_graph_reader.cpp

```diff
@@ -48,7 +48,7 @@ void load_rr_file(const t_graph_type graph_type,
     auto& device_ctx = g_vpr_ctx.mutable_device();
 
     size_t num_segments = segment_inf.size();
-    device_ctx.rr_segments.reserve(num_segments);
+    device_ctx.rr_graph_builder.reserve_segments(num_segments);
     for (long unsigned int iseg = 0; iseg < num_segments; ++iseg) {
         device_ctx.rr_segments.push_back(segment_inf[(iseg)]);
```

Comment:
@tangxifan 
How we are going to replace `device_ctx.rr_segments.push_back(segment_inf[(iseg)]);`.
Because If I used device_ctx.add_rr_segments.(segment_inf[(iseg)]); unit test failed.

Reply:
@umariqbal-rs Please refer to the example here:
https://github.com/lnis-uofu/OpenFPGA/blob/aa8d469fdc221930f0a4f6d45fe7f56f086239b8/vpr/src/route/rr_graph.cpp#L603-L605